src/wui/attack_box.cc

```diff
@@ -29,43 +29,46 @@
 #include "graphic/text_layout.h"
 #include "logic/map_objects/tribes/militarysite.h"
 #include "logic/map_objects/tribes/soldier.h"
+#include "map_io/map_object_loader.h"
+#include "map_io/map_object_saver.h"
+#include "wui/game_debug_ui.h"
+#include "wui/interactive_player.h"
 
 constexpr Duration kUpdateTimeInGametimeMs = Duration(500);  //  half a second, gametime
-
-AttackBox::AttackBox(UI::Panel* parent,
-                     Widelands::Player* player,
-                     Widelands::FCoords* target,
-                     uint32_t const x,
-                     uint32_t const y)
-   : UI::Box(parent, UI::PanelStyle::kWui, x, y, UI::Box::Vertical),
-     player_(player),
-     map_(player_->egbase().map()),
+constexpr int kSpacing = 8;
+
+AttackBox::AttackBox(InteractivePlayer& parent,
+                     UI::UniqueWindow::Registry& reg,
+                     const Widelands::Coords& target,
+                     bool fastclick)
+   : UI::UniqueWindow(&parent, UI::WindowStyle::kWui, "attack", &reg, 0, 0, _("Attack")),
+     iplayer_(parent),
+     map_(iplayer_.player().egbase().map()),
      node_coordinates_(target),
```

Comment:
maybe rename to target_coordinates? I think this is what it is, right?
And not sure if I miss something, but wouldn't it be better to (maybe additionally) pass the building itself so we avoid multiple 
`upcast(Widelands::Building, building, iplayer_.egbase().map().get_immovable(node_coordinates_))` throughout this file? The caller in InteractivePlayer::show_attack_box already upcasts the coords to Building...